airbyte-cdk/python/airbyte_cdk/sources/streams/partitions/partition_generator.py

```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from abc import ABC, abstractmethod
+from typing import Iterable
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.partitions.partition import Partition
+
+
+class PartitionGenerator(ABC):
+    @abstractmethod
+    def generate(self, sync_mode: SyncMode) -> Iterable[Partition]:
+        # FIXME: Pass the state here when add support for incremental
```

Comment:
Why is the state relevant here? Haven't we decided that it shouldn't be used for interpolation? What other use is there?

Reply:
The state won't be used for interpolation, but to generate the partititions.

When running a sync with a state saying all the data prior to 2021-01-01 was synced, we should only generate partitions later than that date